autopts/rtt.py

```diff
@@ -57,14 +59,22 @@ def _get_buffer_index(self, buffer_name):
             if buf.name == buffer_name:
                 return buf_index
 
-    @staticmethod
-    def _read_from_buffer(jlink, buffer_index, stop_thread, user_callback, user_data):
+    def _read_from_buffer(self, jlink, buffer_index, stop_thread, user_callback, user_data):
+        count = 0
+        threshold = 5
         try:
             while not stop_thread.is_set() and jlink.connected() and not get_global_end():
                 byte_list = jlink.rtt_read(buffer_index, 1024)
                 if len(byte_list) > 0:
+                    count = 0
                     user_callback(bytes(byte_list), user_data)
-        except BaseException:
+                elif self.test_end:
+                    count += 1
+                    if count >= threshold:
+                        # we have reached the threshold for consecutive empty reads, we can stop rtt_read
+                        self.rtt_stop = True
```

Comment:
Seems to me that this loop of checking for 5 iterations may be too fast. 
Should there be a delay in here as well?

Reply:
We will enter this loop only after timeout provided with -rtt_logger_param is reached, so maybe incrementing it will help?

Reply:
Got rid of 5 iterations check. Instead,  Timer will be started when first empty packet is received and then cancelled if non-empty packet is received.